crates/rsonpath-lib/src/query/automaton.rs

```diff
@@ -18,8 +18,89 @@ pub struct Automaton<'q> {
     states: Vec<StateTable<'q>>,
 }
 
+/// Represent the distinct methods of moving on a match between states.
+#[derive(Debug, Copy, PartialEq, Clone, Eq)]
+pub enum TransitionLabel<'q> {
+    /// Wraps a textual field [`Label`] in a JSON object.
+    ObjectMember(&'q Label),
+    /// Wraps an array index [`NonNegativeArrayIndex`] in a JSON object.
+    ArrayIndex(NonNegativeArrayIndex),
+}
+
+impl<'q> TransitionLabel<'q> {
+    /// Return a display object with a UTF8 representation of this label.
+    #[must_use]
+    #[inline(always)]
+    pub fn display(&self) -> impl Display + 'q {
+        match self {
+            // This isn't just calling label.display() because the compile makes opaque types unique at the declaration site.
+            // So, the one for display() that is defined above would not equate to Label's def.
+            TransitionLabel::ObjectMember(label) => format!("{}", label.display()),
+            TransitionLabel::ArrayIndex(index) => format!("{index}"),
+        }
+    }
```

Comment:
What's the reason for a dedicated method over just implementing `Display`?

Reply:
This is a refactoring artifact from attempting to get `Label` and `TransitionLabel` to have similiar `display` signatures.  It has been factored out.

Reply:
This is a refactoring artifact from attempting to get `Label` and `TransitionLabel` to have similar `display` signatures.  It has been factored out.